### ISSUE TITLE
Don't read internal files from postcss

### ISSUE THREAD

**Issue by RyanZim** [postcss/autoprefixer#980]
Currently autoprefixer does things like [`const list = require('postcss/lib/list');`](https://github.com/postcss/autoprefixer/blob/master/lib/hacks/cross-fade.js#L2) 
instead of `const list = require('postcss').list;`

This makes bundling autoprefixer, postcss, and other postcss plugins with bundle splitting very difficult to do without duplicating parts of postcss.

Additionally, it could be argued this is breaking [point 2.4 the plugin guidelines](https://github.com/postcss/postcss/blob/master/docs/guidelines/plugin.md#24-use-only-the-public-postcss-api), since the paths aren't documented in the API docs. This prevents postcss from changing its internal directory structure.

Would you be open to changing this?

**Comment by ai** [comment]
Why it makes it difficult to split bundle?

Anyway we have `postcss.list = require ("postcss/list")`

**Comment by RyanZim** [comment]
> Why it makes it difficult to split bundle?

Not sure about all bundler, but some (like browserify) treat each file as its own module. So if you specify that `postcss` should be split to a different bundle file, `postcss/lib/list` won't be split out. The end result is that the `list` code is included in both bundle files.

> Anyway we have postcss.list = require ("postcss/list")

Not sure I understand?

**Comment by ai** [comment]
There is no 1 `postcss` bundle in PostCSS npm package. I don't use Rollup or webpack during npm publish. So PostCSS will be a bunch of small files in the bundle.

Just open `node_modules/postcss/`.

So anyway your bundle will have separated file for `postcss/list`. What all be a difference if I will use `postcss.list` in Autoprefixer?

**Comment by RyanZim** [comment]
I know it isn't one big file. However, if you only require `postcss` directly, bundlers are able to figure out what you're doing and be smarter about code splitting.

i.e. if bundle A contains `postcss`, and bundle B contains `autoprefixer`, the bundler can see that `autoprefixer` can use `postcss` from bundle A. However, it doesn't know that `postcss/lib/list` can be obtained from `postcss.list` as well, so it it includes the `list` code in bundle B as well.

**Comment by RyanZim** [comment]
TL;DR it makes my life a lot easier. Is there a specific reason we _can't_ do `const list = require('postcss').list;`?

**Comment by ai** [comment]
Yeap, it is not big deal, I can replace it. Do you want to send PR?

**Comment by RyanZim** [comment]
Sure, I can do that.
